extensions/vertx-http/dev-ui-resources/src/main/resources/dev-ui/qwc/qwc-build-steps.js

```diff
@@ -1,29 +1,135 @@
-import { LitElement, html, css} from 'lit';
-import { buildSteps } from 'devui-data';
+import { LitElement, html, css } from 'lit';
+
+import { JsonRpc } from 'jsonrpc';
+import { until } from 'lit/directives/until.js';
+import '@vaadin/grid';
+import { columnBodyRenderer } from '@vaadin/grid/lit.js';
+import '@vaadin/grid/vaadin-grid-sort-column.js';
+import '@vaadin/icon';
+import '@vaadin/text-field';
+import '@vaadin/vertical-layout';
+import '@vaadin/horizontal-layout';
 
 /**
  * This component shows the Build Steps
  */
 export class QwcBuildSteps extends LitElement {
+
+  jsonRpc = new JsonRpc(this, true, 'build-metrics');
+
   static styles = css`
-        .todo {
-            padding-left: 10px;
-            height: 100%;
-        }`;
+      .build-steps {
+        height: 100%;
+        display: flex;
+        flex-direction: column;
+        overflow: hidden;
+      }
+
+      vaadin-grid {
+        height: 100%;
+      }
+
+      vaadin-grid-cell-content {
+        vertical-align: top;
+        width: 100%;
+      }
+
+      .summary {
+        margin-bottom: 15px;
+      }`;
 
   static properties = {
-    _steps: {state: true}
+    _buildStepsMetrics: { state: true },
+    _filtered: {state: true, type: Array}
   };
-  
+
   constructor() {
     super();
-    this._steps = buildSteps;
+    this.jsonRpc.getBuildStepsMetrics().then(e => {
+      this._buildStepsMetrics = e.result;
+      this._filtered = this._buildStepsMetrics.records;
+    });
   }
 
   render() {
-    if(this._steps){
-      return html`<div class="todo">${this._steps}</div>`;
+    return html`${until(this._render(), html`<span>Loading build steps...</span>`)}`;
```

Comment:
AFAICS, this actually doesn't work. The first loading should be slow and I don't see the loading message.

Reply:
Yes I have noticed it too. See https://github.com/quarkusio/quarkus/blob/6048280a15e2bc25ed611a35c34b5217f5d83b08/extensions/kubernetes/vanilla/deployment/src/main/resources/dev-ui/qwc-kubernetes-manifest.js#L41 on how to implement this with a property

Reply:
I changed this in my commit